vspace/src/main/java/edu/asu/diging/vspace/web/staff/ListImagesController.java

```diff
@@ -48,7 +50,19 @@ public String listSpaces(@PathVariable(required = false) String page,
         model.addAttribute("totalPages", imageService.getTotalPages());
         model.addAttribute("currentPageNumber", pageNo);
         model.addAttribute("totalImageCount", imageService.getTotalImageCount());
-        List<IVSImage> images = imageService.getImages(pageNo, sortedBy, order);
+        List<IVSImage> images;
+        if(imageCategory.equals("Show All")) {
+            model.addAttribute("totalPages", imageService.getTotalPages());
+            model.addAttribute("currentPageNumber", pageNo);
+            model.addAttribute("totalImageCount", imageService.getTotalImageCount());
+            images = imageService.getImages(pageNo, sortedBy, order);
+        }
+        else {
+            model.addAttribute("totalPages", imageService.getTotalPagesForSelectedFilter(ImageCategory.valueOf(imageCategory)));
```

Comment:
since you're passing the category here into the method `getTotalPages` is descriptive enough as a method name

Reply:
javadoc should do the rest of the explaining